### PR TITLE
Remove dtbo_block_device which is defined in AOSP since QPR2

### PR DESCRIPTION
**Pull request by mlehtima**
Ref Repo : https://android.googlesource.com/platform/system/sepolicy/+/refs/tags/android-14.0.0_r30/public/device.te#99
Ref Commit : https://android.googlesource.com/platform/system/sepolicy/+/5d0ef8448f165bdfe59b01d822798a3276c6ca09

Error : device/sony/sepolicy/vendor/device.te:30:ERROR 'Duplicate declaration of type' at token ';'
 on line 90868:
type dtbo_block_device, dev_type;

This change was originally added in https://github.com/sonyxperiadev/device-sony-sepolicy/pull/610 but got lost when recent changes were pushed.

### CODE REVIEW

**Review by MarijnS95 — CHANGES_REQUESTED** [review]
> This change was originally added in https://github.com/sonyxperiadev/device-sony-sepolicy/pull/610 but got lost when recent changes were pushed.

That's good because the original PR didn't have a correct/useful commit message, and the [request to go one step further](https://github.com/sonyxperiadev/device-sony-sepolicy/pull/610#issuecomment-2082650372) and also delete our duplicate rule on `allow init dtbo_block_device ...  relabelto;`, wasn't followed through either. Can you please do that before we merge this?

**Comment by mlehtima** [comment]
@MarijnS95  Just to be sure that you want me to remove https://github.com/sonyxperiadev/device-sony-sepolicy/blob/master/vendor/init.te#L11 and also remove the blk_file from line "}:{ blk_file lnk_file } relabelto;" i.e. change it to "}:lnk_file relabelto;"? Anything else needed?

**Comment by MarijnS95** [comment]
@mlehtima hmm, `boot_block_device` only has `lnk_file` upstream. If we label a `blk_file` with this _file context_ we should definitely keep `blk_file` in there. `vbmeta_block_device` isn't defined by AOSP at all, so we should equally check if this _file context_ is used on a block device or a link file, and update the line accordingly.

When grepping for `boot_block_device`/`boot_block_device`/`vbmeta_block_device`, our device repositories are only setting this file context on symlinks, so it should be good to only keep `relabelto` for `blk_file`.

TLDR: please test if AOSP `restorecon -R` in `init` still works, but I think this should be the end result:

```te
allow init vbmeta_block_device:lnk_file relabelto;
```

**Comment by mlehtima** [comment]
I didn't find that boot_block_device mentioned with relabelto in system/sepolicy, where did you find it? Should we still keep that together with vbmeta_block_device?

**Comment by MarijnS95** [comment]
https://cs.android.com/search?q=boot_block_device&sq= @mlehtima correct, I read a bit too quickly and didn't see that it was device-specific policy, good cathc. Let's keep it in the list, so:

```te
allow init { boot_block_device vbmeta_block_device }:lnk_file relabelto;
```

**Comment by mlehtima** [comment]
@MarijnS95 I updated the changes now and also partially rewrote the commit message

File: vendor/init.te [hunk 1]
```diff
@@ -5,12 +5,8 @@ allow init {tmpfs configfs }:lnk_file create_file_perms;
 allow init { mnt_vendor_file vendor_firmware_file adsprpcd_file }:dir mounton;
 allow init vendor_firmware_file:filesystem { mount relabelfrom };
 
-# AOSP init only covers system_block_device, add our own:
-allow init {
-    boot_block_device
-    dtbo_block_device
-    vbmeta_block_device
-}:{ blk_file lnk_file } relabelto;
+# Allow init to relabel boot and vbmeta block devices
```
Comment (MarijnS95) [inline vendor/init.te:8]:
```suggestion
# Allow init to relabel symlinks to boot and vbmeta block devices
```

**Review by MarijnS95 — COMMENTED** [review]
I still don't like the original wording and formatting but at least the extra cleanup is documented now.

File: vendor/init.te (hunk 1 above)
Comment (mlehtima) [reply vendor/init.te:8]: I rewrote the commit message a bit more, hopefully it's better now.

Comment (MarijnS95) [reply vendor/init.te]: @mlehtima I'd suggest rewriting `symlinks dtbo block device` -> `` symlinks to `dtbo_block device` `` for grammar purposes, but other than that this is _exactly_ what I was looking for, thanks!

Feel free to keep the build error in though, that's not harmful and a good demonstration.

Comment (mlehtima) [reply vendor/init.te]: Done

**Review by MarijnS95 — APPROVED** [review]
